### PR TITLE
feat: ✨ allow duplicate team names with a unique slug

### LINKED ISSUE #2276: feat: ✨ allow duplicate team names with a unique slug
## Problem

`Team.name` is `@unique`, so two teams cannot share a name. In practice this causes conflicts (distinct organizations legitimately want the same display name), and a duplicate currently surfaces as a generic **500** on team creation with no clear message.

## Solution

Move the uniqueness constraint from `name` to a new **auto-generated `slug`**:

- `Team.name` is no longer unique — teams may be homonyms.
- Add `Team.slug String @unique`, derived from the name on creation (`slugify` + numeric suffix on collision: `acme-corp`, `acme-corp-2`, …). Users never enter or see it.
- Scope: **uniqueness only** — routing stays by numeric `id`; the API just returns the new `slug` field.
- Migration backfills slugs for existing teams (with a safety pass that guarantees uniqueness before the unique index).

## Out of scope

- Slug-based URLs (`/teams/acme-corp`) — the slug is stored and returned but routing is unchanged.
- User-editable slugs.

### PR DESCRIPTION
**Pull request by hermannleboss**
Closes #2276

## What

Teams can now share a display **name**; uniqueness moves to a new, auto-generated **`slug`**.

- **Schema**: `Team.name` drops `@unique`; add `Team.slug String @unique`.
- **Slug generation**: `slugify(name)` + numeric suffix on collision (`acme-corp`, `acme-corp-2`, …), wired in `addTeam` and the seeder. The util (`backend/src/utils/slug.util.ts`) is a dependency-free pure function (`slugify` + `generateUniqueSlug(base, exists)`), so the store check is injected and trivially unit-testable.
- **Race guard**: on a `P2002` between the uniqueness check and the insert, the controller regenerates the slug and retries once.
- **Migration** (`20260710130001_team_unique_slug`): drops the name unique index, backfills slugs from existing names (de-duping homonyms), then a safety pass appends the row `id` to any residual cross-partition collision **before** creating the unique index, so it never aborts on real data.
- **Frontend**: `slug` added to the `Team` type; removed the now-inaccurate "unique" wording from the create-company name hint. Routing is unchanged (still by numeric `id`).

## Scope

Uniqueness only. Slug-based URLs and user-editable slugs are intentionally out of scope.

## Verification

- Backend unit suite: **666 passing** (incl. new `slug.util` tests and two `addTeam` slug tests).
- App: `type-check` clean, unit suite **2258 passing**.
- Migration backfill run against an ephemeral Postgres with edge cases (homonyms, symbol-only names, the `Acme` / `Acme 2` cross-partition collision, accents) — all slugs unique, `CREATE UNIQUE INDEX` succeeds.

> Apply the migration (`prisma migrate deploy`) in each environment on merge.

### CODE REVIEW

File: backend/src/utils/slug.util.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * `'team'` when the input has no usable characters (e.g. only symbols).
+ *
+ * The output always satisfies the shared `slugSchema` regex
+ * (`^[a-z0-9]+(?:-[a-z0-9]+)*$`) in `validation/utils.ts`.
+ */
+export const slugify = (input: string): string => {
+  const slug = input
+    .toLowerCase()
+    .replace(/[^a-z0-9]+/g, '-')
+    .replace(/^-+|-+$/g, '');
```
[the remark this answers is not in the document]
Comment (hermannleboss) [reply backend/src/utils/slug.util.ts]: Fixed in fb49fe5: replaced `/^-+|-+$/` with `/^-|-$/` (no `+` quantifier). The preceding `.replace(/[^a-z0-9]+/g, "-")` already collapses every non-alphanumeric run to a single hyphen, so there are never consecutive hyphens to trim — matching a single leading/trailing hyphen is equivalent and stays linear, eliminating the polynomial-regex class.
